multichain-aggregator/multichain-aggregator-logic/src/search.rs

```diff
@@ -0,0 +1,58 @@
+use crate::{
+    error::ServiceError,
+    repository::{addresses, block_ranges, hashes},
+    types::{
+        chains::Chain,
+        search_results::{ChainSearchResult, SearchResults},
+    },
+};
+use sea_orm::DatabaseConnection;
+use std::collections::BTreeMap;
+use tokio::try_join;
+
+macro_rules! populate_search_results {
+    ($target:expr, $explorers:expr, $from:expr, $field:ident) => {
+        for e in $from {
+            let chain_id = e.chain_id.to_string();
```

Comment:
why to string?

Reply:
why cannot we store type `ChainId` instead of `String` in `SearchResults`

Reply:
Agree. Changed to `ChainId` for internal types and added cast to string only for proto struct.